packages/core/src/linker/query_list.ts

```diff
@@ -49,8 +49,8 @@ export class QueryList<T> implements Iterable<T> {
   private _changes: EventEmitter<QueryList<T>>|null = null;
 
   readonly length: number = 0;
-  readonly first: T = undefined!;
-  readonly last: T = undefined!;
+  readonly first: T|undefined = undefined;
+  readonly last: T|undefined = undefined;
```

Comment:
Is the `= undefined` assingment still needed? I believe that it could be removed, no?

```suggestion
  readonly first?: T;
  readonly last?: T;
```

Reply:
yeah good call, thanks :slightly_smiling_face: